client/pages/sign-up.jsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+
+export default class SignUp extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      firstName: '',
+      lastName: '',
+      username: '',
+      password: '',
+      email: ''
+    };
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleChange = this.handleChange.bind(this);
+  }
+
+  handleChange(event) {
+    const { name, value } = event.target;
+    this.setState({ [name]: value });
+  }
+
+  handleSubmit(event) {
+    event.preventDefault();
+    const req = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(this.state)
+    };
+    fetch('/api/auth/sign-up', req)
+      .then(res => res.json())
+    // eslint-disable-next-line no-console
+      .catch(err => console.log(err));
```

Comment:
When you are catching promise rejections, you can use `console.error(err)` instead of `console.log()`. That way you can delete the `// eslint-disable...`